sotodlib/toast/scripts/so_batch_control.py

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2024-2024 Simons Observatory.
+# Full license can be found in the top level "LICENSE" file.
+
+"""
+Helper script for batching observations.
+
+The results are printed to stdout to enable use of this
+script inside a slurm script.
+
+"""
+
+import argparse
+import datetime
+import os
+import re
+import sys
+
+import numpy as np
+
+from astropy import units as u
+
+# Import sotodlib.toast first, since that sets default object names
+# to use in toast.
+import sotodlib.toast as sotoast
+
+import toast
+import toast.ops
+from toast.mpi import MPI, Comm
+from toast.observation import default_values as defaults
+
+from .. import ops as so_ops
+from .. import workflows as wrk
+
+
+def load_obs_file(path):
+    """Load the observation file into a list.
+
+    Args:
+        path (str):  Path to the file
+
+    Returns:
+        (list):  The list of observation IDs.
+
+    """
+    olist = list()
+    with open(path, "r") as f:
+        for line in f:
+            if re.match(r"^#.*", line) is None:
+                olist.append(line.strip())
+    return olist
+
+
+def get_obs_state(out_root, obs):
+    """Find the current processing state of an observation.
+    """
+    obs_dir = os.path.join(out_root, obs)
+    state_file = os.path.join(obs_dir, "state")
+    if os.path.isfile(state_file):
+        with open(state_file, "r") as f:
+            state = f.readline()
+        return state.rstrip()
+    return None
+
+
+def clear_obs_state(out_root, obs):
+    """Clear the current processing state of an observation.
+    """
+    obs_dir = os.path.join(out_root, obs)
+    if not os.path.isdir(obs_dir):
+        return
+    state_file = os.path.join(obs_dir, "state")
+    if os.path.isfile(state_file):
+        os.remove(state_file)
+
+
+def set_obs_state(out_root, obs, state):
+    """Set the current processing state of an observation.
+    """
+    obs_dir = os.path.join(out_root, obs)
+    if not os.path.exists(obs_dir):
+        os.makedirs(obs_dir)
+    state_file = os.path.join(obs_dir, f"state")
+    temp_state = os.path.join(obs_dir, f"state_temp")
```

Comment:
Should the code check if `state_temp` already exists?

Reply:
Although this temp file should only exist for a fraction of a second, I added a check and raise an exception if it exists.